be/src/storage/inverted/clucene/clucene_inverted_writer.cpp

```diff
@@ -49,6 +49,8 @@ class CLuceneInvertedWriterImpl : public CLuceneInvertedWriter {
         _parser_type = get_inverted_index_parser_type_from_string(
                 get_parser_string_from_properties(_inverted_index->index_properties()));
         _field_name = std::wstring(field_name.begin(), field_name.end());
+        _wstring_buffer = (void*)malloc(2 * 1024 * 1024);
```

Comment:
can this buffer can be allocated as near as its usage? so the mem is not necessary whenever a CLuceneInvertedWriterImpl object is created.

Reply:
This is done intentionally to allocate more memory in advance. The purpose here is to ensure that the buffer is large enough to accommodate the contents in most rows and avoid frequent memory reallocation.

Reply:
that's fine to allocate memory all at once large enough. But it is not necessary to allocate when constructing the object.

E.g. when do things like `auto a = std::vector<CLuceneInvertedWriterImpl>(10)` and then `a.clear()`, these mem allocation is useless. allocating the memory as near as to its usage, make the constructing of the object as lightweight as possible.

Reply:
Fixed it, allocate memory when we use it.

Reply:
refactor the code for correctness.